tests/functionaltests/Tests/UIKitTests/UIButtonTests.mm

```diff
@@ -119,69 +118,1156 @@ - (void)dealloc {
+        UIControlCheckStateTransition(buttonToTest,
+                                      uxEvent,
+                                      UIControlStateHighlighted,
+                                      @selector(currentBackgroundImage),
+                                      [buttonToTest backgroundImageForState:UIControlStateHighlighted]);
+        expectedBackgroundImage = @"125x125.png";
+        UIControlCheckStateTransition(buttonToTest,
+                                      uxEvent,
+                                      UIControlStateSelected,
+                                      @selector(currentBackgroundImage),
+                                      [buttonToTest backgroundImageForState:UIControlStateSelected]);
+    }
+
+    TEST_METHOD(UIButton_TitleLabel) {
+        StrongId<UIButtonWithControlsViewController> buttonVC;
+        buttonVC.attach([[UIButtonWithControlsViewController alloc] init]);
+        UXTestAPI::ViewControllerPresenter testHelper(buttonVC);
+
+        UIButton* buttonToTest = [buttonVC defaultButton];
 
+        WXFrameworkElement* xamlElement = [buttonToTest.titleLabel _getXamlTextBlock];
+        ASSERT_OBJCNE(xamlElement, nil);
+    }
+
+    TEST_METHOD(UIButton_ContentEdgeInsets) {
+        BEGIN_TEST_METHOD_PROPERTIES()
+        TEST_METHOD_PROPERTY(L"ignore", L"true")
+        END_TEST_METHOD_PROPERTIES()
+
+        StrongId<UIButtonWithControlsViewController> buttonVC;
+        buttonVC.attach([[UIButtonWithControlsViewController alloc] init]);
+        UXTestAPI::ViewControllerPresenter testHelper(buttonVC);
+
+        UIButton* buttonToTest = [buttonVC defaultButton];
+
+        __block auto uxEvent = UXEvent::CreateAuto();
+        __block auto xamlWidthSubscriber = std::make_shared<XamlEventSubscription>();
+        __block auto xamlHeightSubscriber = std::make_shared<XamlEventSubscription>();
+        __block auto xamlCanvasLeftSubscriber = std::make_shared<XamlEventSubscription>();
+        __block auto xamlCanvasTopSubscriber = std::make_shared<XamlEventSubscription>();
+
+        WXFrameworkElement* xamlElement = [buttonToTest xamlElement];
+        ASSERT_OBJCNE(xamlElement, nil);
+
+        // TODO: Get contentCanvas child
+
+        dispatch_sync(dispatch_get_main_queue(), ^{
+            [buttonVC textImageStateField].text = @"N;";
+            [buttonVC textImage].text = @"50x50.png";
+
+            [buttonVC textTitleStateField].text = @"N;";
+            [buttonVC textTitle].text = @"Normal";
+            [buttonVC textTitleColorStateField].text = @"N;";
+            [buttonVC textTitleColor].text = @"redColor";
+
+            // Register RAII event subscription handler
+            xamlWidthSubscriber->Set(xamlElement, [WXCCanvas widthProperty], ^(WXDependencyObject* sender, WXDependencyProperty* dp) {
+                double width = DoubleFromPropertyValue([sender getValue:dp]);
+
+                // Validation
+                EXPECT_EQ(width, 30.0f); // now 0
```

Comment:
I was looking at contentCanvas under VS's Live Property Explorer. I've disabled tests for ContentEdgeInsets and TitleEdgeInsets after the last merge as these tests need to be updated and I'd like understand if I'm looking at the right values.

---
In reply to: [101126737](https://github.com/Microsoft/WinObjC/pull/1947#discussion_r101126737) [](ancestors = 101126737)